app/controllers/application_controller.rb

```diff
@@ -46,6 +46,7 @@ def handle_unverified_request
   before_action :clear_notifications
   before_action :set_locale
   before_action :set_mobile_view
+  before_action :handle_delegated_auth
```

Comment:
can we avoid adding an extra `before action` I am pretty protective of having too many ? maybe just do this direct in current user provider ?